### ISSUE TITLE
num_classes in deeplabv3_resnet50

### ISSUE THREAD

**Issue by claudiotancredi** [pytorch/vision#7430]
Hi, I'm having trouble figuring out which is the correct number of classes that I have to use to build the DeepLabV3 model with a ResNet-50 backbone.
I'm working on the Cityscapes dataset and I'm re-mapping the original classes to 19 classes, using:
`mapping = {
    0: 255,
    1: 255,
    2: 255,
    3: 255,
    4: 255,
    5: 255,
    6: 255,
    7: 0,
    8: 1,
    9: 255,
    10: 255,
    11: 2,
    12: 3,
    13: 4,
    14: 255,
    15: 255,
    16: 255,
    17: 5,
    18: 255,
    19: 6,
    20: 7,
    21: 8,
    22: 9,
    23: 10,
    24: 11,
    25: 12,
    26: 13,
    27: 14,
    28: 15,
    29: 255,
    30: 255,
    31: 16,
    32: 17,
    33: 18,
    -1: 255}`
Now I have classes from 0 to 18 and the 255 class for background. 
I've worked on Cityscapes before and I've always trained my models passing `num_classes=19`, but the comments on the DeepLabV3 code state:
https://github.com/pytorch/vision/blob/b403bfc771e0caf31efd06d43860b09004f4ac61/torchvision/models/segmentation/deeplabv3.py#L255
Therefore, I don't understand if num_classes should be 19 or 20, in my case. Anyway, I'm using a cross entropy loss with `ignore_index=255`. Do you have suggestions, please?

**Comment by NicolasHug** [comment]
Hi @claudiotancredi , if there are 19 classes in the dataset (excluding the background) then you should be passing `num_classes=20`.

> I've always trained my models passing num_classes=19

What models are you referring to?

**Comment by claudiotancredi** [comment]
@NicolasHug I'm sorry, my mistake. I used BiSeNet in the past but I adapted the code to pass 19 as `num_classes`, so thank you.
I have another question, if I may. I'm currently trying to use DeepLabV3 model with a ResNet-50 backbone on Cityscapes, but it seems I'm unable to achieve a validation mIoU higher than 60%, so I'm not able to reach the performance given in the [official paper](https://arxiv.org/pdf/1706.05587v3.pdf). 
My settings are:

- batch size = 8
- image size = 769
- cross entropy loss
- different learning rates ranging from 1e-3 to 1e-4
- weight decay = 1e-4 or 1e-5
- optimizer = adamw
- no augmentation
- no pre-training

I'm also using PyTorch Lightning's Trainer. 
I've tried working with different batch sizes, different learning rates, different values for weight decay, different optimizers, thinking that maybe my model only needed to be tuned a little bit, but the validation mIoU is always around 60%. What do you think it could be the issue? Does this setting seem appropriate to you, if I want to train the model from scratch?

**Comment by NicolasHug** [comment]
Sorry @claudiotancredi I'm not too familiar with how the authors trained their models on CityScape, and I won't have more input than what's already written in the paper. For that kind of question, it's preferable to direct them to https://discuss.pytorch.org/ rather than on this issue tracker, which is reserved for bug report and feature request.

Since the original question has been answered I'll close the issue. Thanks!
